ocaml/xapi/import.ml

```diff
@@ -2028,7 +2078,10 @@ let with_open_archive fd ?length f =
   try
     Tar_unix.really_read fd buffer ;
     (* we assume the first block is not all zeroes *)
-    let hdr = Option.get (Tar.Header.unmarshal buffer) in
+    let hdr =
+      TarHeaderReader.read (BufferOrFile.make retry_with_compression buffer fd)
+      |> Result.get_ok
+    in
     assert_filename_is hdr ;
     (* successfully opened uncompressed stream *)
     retry_with_compression := false ;
```

Comment:
Is this the reason why a ref variable is used in the `in_channel` record rather than a mutable record?

Reply:
it is set from 2 different places, a mutable record field could've been used but there was no record here previously, just local values.